src/main/java/com/rackspacecloud/metrics/ingestionservice/influxdb/providers/ProdTenantRouteProvider.java

```diff
@@ -16,17 +16,12 @@
      * @param measurement
      * @param restTemplate is used to connect to the routing service to get the route
      * @return
-     * @throws Exception
      */
     @Override
-    public TenantRoutes getRoute(String tenantId, String measurement, RestTemplate restTemplate) throws Exception {
+    public TenantRoutes getRoute(String tenantId, String measurement, RestTemplate restTemplate) {
         String requestUrl = String.format("%s/%s/%s", tenantRoutingServiceUrl, tenantId, measurement);
 
-        try {
-            return restTemplate.getForObject(requestUrl, TenantRoutes.class);
-        }
-        catch (Exception e) {
-            throw new Exception(String.format("Exception thrown for requestUrl [%s]", requestUrl), e);
-        }
```

Comment:
Part of the original strategy was good in that it insulates the caller from knowledge of the `RestClientException` that could be thrown by [getForObject](https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/web/client/RestTemplate.html#getForObject-java.net.URI-java.lang.Class-). It might be good to catch that specifically and throw a new `ExternalSystemException` (for example). If a REST API were exposed to let users ingest metrics (like blueflood), then that exception can be mapped to a 502 response.